peer/brontide.go

```diff
@@ -1050,6 +1057,14 @@ func (p *Brontide) maybeSendNodeAnn(channels []*channeldb.OpenChannel) {
 // calling Disconnect will signal the quit channel and the method will not
 // block, since no goroutines were spawned.
 func (p *Brontide) WaitForDisconnect(ready chan struct{}) {
+	// Before we try to call the `Wait` goroutine, we'll make sure the main
+	// set of goroutines are already active.
+	select {
+	case <-p.startReady:
+	case <-p.quit:
+		return
+	}
```

Comment:
So this part is actually redundant... If we never `Disconnect` before we're done with `Start`, then `p.quit` won't be closed before `startReady`.

Reply:
I think it's fine belt-and-suspenders even if it's redundant